irods/test/data_obj_test.py

```diff
@@ -834,59 +834,60 @@ def do_test_redirect_in_data_object_put_and_get__issue_452(self, content, data_c
-                srch(PUT_LOG)
-                generator = None
-                # Activate a read ticket on a new session if necessary, and attempt a GET
-                if data_ctx['ticket_access']:
-                    for access in iRODSAccess('own',remote_name,self.sess.username), \
-                                  iRODSAccess('null',remote_name,sess.username):
-                        self.sess.acls.set(access, admin = True)
-                    generator = self._data_object_and_associated_ticket(data_name=remote_name, auto_delete_data = False, ticket_access='read')
-                    # Emulate the 'with'-block construction for the read ticket:
-                    data_ctx_get = next(generator)
-                    data_ctx_get['initialize']()
-                    sess = data_ctx_get['session']
-                GET_LOG = self.In_Memory_Stream(always_unicode = True)
-                with helpers.enableLogging(logging.getLogger('irods.manager.data_object_manager'),
-                                           logging.StreamHandler, (GET_LOG,), level_ = logging.DEBUG),\
-                     helpers.enableLogging(logging.getLogger('irods.parallel'),
-                                           logging.StreamHandler, (GET_LOG,), level_ = logging.DEBUG):
-                    sess.data_objects.get(remote_name,LOCAL_FILE)
-                srch(GET_LOG)
-                with open(LOCAL_FILE,'rb') as get_result:
-                    self.assertTrue(content, get_result.read())
-                # Finalize the emulated 'with'-block construction for the read ticket, if active:
-                del generator
-                data_ctx['finalize']()
-        finally:
-            if os.path.isfile(LOCAL_FILE):
-                os.unlink(LOCAL_FILE)
-            if filename:
-                os.unlink(filename)
+        with config.loadlines(entries=[dict(setting='data_objects.allow_redirect',value=True)]):
+            try:
+                with self.create_simple_resc(hostname = 'localhost') as rescName:
+                    with NamedTemporaryFile(delete = False) as f:
+                        filename = f.name
+                        f.write(content)
+                    data_ctx['initialize']()
+                    sess = data_ctx['session']
+                    remote_name = data_ctx['path']
+                    PUT_LOG = self.In_Memory_Stream(always_unicode = True)
+                    with helpers.enableLogging(logging.getLogger('irods.manager.data_object_manager'),
+                                               logging.StreamHandler, (PUT_LOG,), level_ = logging.DEBUG),\
+                         helpers.enableLogging(logging.getLogger('irods.parallel'),
+                                               logging.StreamHandler, (PUT_LOG,), level_ = logging.DEBUG):
+                        sess.data_objects.put(filename, remote_name, **{kw.DEST_RESC_NAME_KW: rescName})
+                    def srch(BUF):
```

Comment:
Please add a comment explaining what `srch` is for. That'll help readers of the test understand what's happening quicker.

Reply:
let's use a lot more letters please

Reply:
Can do if you want it.... note however it isn't a relevant part of the diff.  This section appears as a diff only because an extra level of indentation was added due to the insertion of a `with`-block right before it.

How infuriatingly Pythonic, you might say.  And you'd be right.